Jenkinsfile

```diff
@@ -1,75 +1,63 @@
-        }
-    }
-
     stages {
         stage('Checkout') {
             steps {
                 // Checkout the repository
-                git branch: 'main', url: 'https://github.com/your-username/your-repository.git'
+                git branch: 'main', url: 'https://github.com/syeda-farha-2023/Jenkins-project.git'
             }
         }
-        stage('Detect Changes') {
+        stage('Activate Service Account') {
             steps {
-                script {
-                    // Get the list of modified files from the PR
-                    def changedFiles = sh(script: 'git diff --name-only origin/main...HEAD', returnStdout: true).trim().split('\n')
-                    
-                    // Initialize an empty list to store updated cloud functions
-                    def updatedFunctions = []
-
-                    // Detect which cloud functions have been modified
-                    for (file in changedFiles) {
-                        if (file.startsWith('cloud_function1/')) {
-                            updatedFunctions.add('cloud_function1')
-                        } else if (file.startsWith('cloud_function2/')) {
-                            updatedFunctions.add('cloud_function2')
-                        }
-                    }
-
-                    // Remove duplicates
-                    updatedFunctions = updatedFunctions.unique()
-
-                    echo "Updated cloud functions: ${updatedFunctions}"
+                withCredentials([file(credentialsId: 'channelpay-cloud-function-admin-sa', variable: 'SERVICE_ACCOUNT_KEY_PATH')]) {
+                    sh 'gcloud auth activate-service-account --key-file="${SERVICE_ACCOUNT_KEY_PATH}"'
                 }
             }
         }
-
-        stage('Deploy Updated Functions') {
+        stage('Deploy Functions') {
             steps {
                 script {
-                    // Loop through updated functions and deploy each one
-                    for (function in updatedFunctions) {
-                        dir(function) {
-                            sh './deploy.sh'
+                    // List all subdirectories (ignoring the current directory)
+                    def directories = sh(script: 'find . -mindepth 1 -maxdepth 1 -type d', returnStdout: true).trim().split('\n')
+                    for (dir in directories) {
+                        dir(dir) {
+                            // Check if deploy.sh exists and execute it
+                            if (fileExists('deploy.sh')) {
+                                sh './deploy.sh'
+                            } else {
+                                // Use a generic deployment command if no deploy.sh exists
+                                echo "No deploy.sh found in ${dir}, using default gcloud deploy command"
+                                sh "gcloud functions deploy --runtime python39 --trigger-http --allow-unauthenticated"
```

Comment:
In the else, throw error and fail the pipeline. We do not have any default deploy functionality.